Tests/FoundationEssentialsTests/PredicateTests.swift

```diff
@@ -14,9 +14,6 @@
 import TestSupport
 #endif
 
-// Predicate does not back-deploy to older Darwin versions
-#if FOUNDATION_FRAMEWORK || os(Linux) || os(Windows)
-
 @available(macOS 14, iOS 17, tvOS 17, watchOS 10, *)
```

Comment:
FYI `@available` does not prevent tests from running when using XCTest. You'll need a guard in each test method, e.g.:

```swift
func testFoo() throws {
  guard #available(macOS 14, iOS 17, tvOS 17, watchOS 10, *) else {
    throw XCTSkip("Unavailable")
  }

  ...
}
```

Reply:
Thanks for the tip, I was just diving into the CI logs to find out why these tests were still running. Instead of guarding within every test, is there a way I can guard at the test case level? Is there perhaps something in `XCTestCase` that is safe to override where I can write this guard and have it apply to every test in the test case?

Reply:
I added this check to `setUp() async throws` to only check once, and moved the availability annotation from the test class to each individual function to avoid a warning for usage of `#if available(macOS 14, ...)` within an `@available(macOS 14, ...)` class.

Reply:
That'll work.